static/app/components/hovercard.tsx

```diff
@@ -79,12 +79,21 @@ interface HovercardProps {
 function Hovercard(props: HovercardProps): React.ReactElement {
   const [visible, setVisible] = useState(false);
 
-  const inTimeout = useRef<number | null>(null);
   const scheduleUpdateRef = useRef<(() => void) | null>(null);
 
   const portalEl = useMemo(() => findOrCreatePortal(), []);
   const tooltipId = useMemo(() => domId('hovercard-'), []);
 
+  const showHoverCardTimeoutRef = useRef<number | null>(null);
+
+  useEffect(() => {
+    return () => {
+      if (showHoverCardTimeoutRef.current) {
+        window.clearTimeout(showHoverCardTimeoutRef.current);
+      }
+    };
+  }, []);
```

Comment:
Yeah, 100%. I went with null because a lot of them were already typed as null, so I consolidated them to be all the same. I'll make a separate PR to rename all this and we can have the discussion there, but I agree with you :)